### ISSUE TITLE
our improve

### ISSUE THREAD

**Issue by yeanoor** [yeanoor/CVPR-final-1#1]
import cv2
import tensorflow as tf
import matplotlib.pyplot as plt

# Load a pre-trained object detection model (e.g., MobileNetV2)
model = tf.keras.applications.MobileNetV2(weights='imagenet', include_top=True)

# Load an image for testing
image_path = 'path/to/your/image.jpg'
image = cv2.imread(image_path)
image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)

# Preprocess the image
input_array = tf.keras.applications.mobilenet_v2.preprocess_input(tf.image.resize(image, (224, 224))[tf.newaxis, ...])

# Get model predictions
predictions = model.predict(input_array)

# Get the top prediction (you may need to customize this based on your model)
predicted_class = tf.argmax(predictions[0])

# Get the feature map of the selected layer for visualization (customize as needed)
layer_name = 'block_5_expand_relu'
intermediate_layer_model = tf.keras.Model(inputs=model.input, outputs=model.get_layer(layer_name).output)
intermediate_output = intermediate_layer_model(input_array)

# Calculate the saliency map based on the intermediate feature map
saliency_map = tf.reduce_max(intermediate_output, axis=-1)

# Normalize the saliency map for better visualization
saliency_map = (saliency_map - tf.reduce_min(saliency_map)) / (tf.reduce_max(saliency_map) - tf.reduce_min(saliency_map))

# Display the original image and saliency map
plt.figure(figsize=(10, 5))
plt.subplot(1, 2, 1)
plt.imshow(image)
plt.title('Original Image')

plt.subplot(1, 2, 2)
plt.imshow(saliency_map[0], cmap='viridis')
plt.title('Saliency Map')

plt.show()
